### ISSUE TITLE
Suggestion: prop/flexbody changes in skinzips

### ISSUE THREAD

**Issue by WillMichals** [RigsOfRods/rigs-of-rods#2736]
Hi all, I was thinking of the idea of incorporating prop or flexbody changes into skinzips.
I thought of two ways to do this; replace a props/flexbody that has already been placed (similar to replacing textures) or adding props that do not exist in the props/flexbody section of the truck but in the skinzip instead.

```
AddPropSkin
{
    addProp = 1, 2, 3, .5, .5, .5, 0, 90, 90, bumper.mesh
    preview          = preview.png
    description      = livery replacement
    authorName       = John Doe
    AuthorID         = 12345
    GUID             = xxx
}
```

_or_

```
ReplacePropSkin
{
    replaceProp = spoiler1.mesh, spoiler2.mesh
    preview          = preview.png
    description      = livery replacement
    authorName       = John Doe
    AuthorID         = 12345
    GUID             = xxx
}
```

**Comment by ohlidalp** [comment]
Thanks for the bump in Discord#development, this deserves a brief thought:
* `replaceProp` the way you suggest it is trivial to code and introduces no duplicities.
* `addProp` is easy to code but needs either hacking the truckfile parser to parse prop-line from external source, or duplicate logic to parse propline. It's still minimal code though so no biggie, just saying.
* But then what?
* Since we have https://github.com/RigsOfRods/rigs-of-rods/pull/3003, modders would probably stop defining props in truckfiles altogether and rely on bundled skins to do it.
* Until now prop customizations were done by `sectionconfig`, which is a powerful feature [when working correctly](https://github.com/RigsOfRods/rigs-of-rods/issues/2881), but presently it's reduced to mutually exclusive sections, so for the purpose of props/flexbodies it's essentially the same.

TL/DR: I'll do this, it can't do any harm.

**Comment by CuriousMike56** [comment]
Now that we have the addon parts system, skins could receive an option to load `.addonpart` files. Selecting a police skin would automatically equip lightbars for example.

**Comment by ohlidalp** [comment]
@CuriousMike56 
> Now that we have the addon parts system, skins could receive an option to load .addonpart files. Selecting a police skin would automatically equip lightbars for example.

I'd rather do it the other way around - give the .tuneup format ability to apply a skin, and then distribute the .tuneup file as an one-click police customization package (skin + addonparts) for example.

I'm going to close this as resolved; this needs further debate (i.e. addonparts vs. sectionconfig) and deserves a fresh ticket.
